src/atomic/atomic.lisp

```diff
@@ -36,18 +44,20 @@
   cell)
 
 (defun make-atomic-integer (&key (value 0))
-  (%make-atomic-integer value))
+  #+ccl (%make-atomic-reference (make-array 1 :initial-element value))
+  #-ccl (%make-atomic-reference (cons value nil)))
 
 (defmethod print-object ((int atomic-integer) stream)
   (print-unreadable-object (int stream :type t :identity t)
     (format stream "~S" (atomic-get int))))
 
 (defmethod atomic-get ((int atomic-integer))
-  (slot-value int 'cell))
+  (atomic-place int))
 
 (defmethod atomic-cas ((int atomic-integer) expected new)
-  (let* ((*the-int* (slot-value int 'cell))
-         (cas-result (atomics:cas *the-int* expected new)))
-    (when cas-result
-      (setf (slot-value int 'cell) *the-int*))
-    cas-result))
+  (atomics:cas (atomic-place int) expected new))
+
+(defmethod atomic-setf ((int atomic-integer) new)
```

Comment:
This new method should also be implemented for ABCL.

Reply:
`atomics` library does not support `abcl`, I will have a try after I install `abcl`.

Reply:
There is a special file `atomic-abcl.lisp` that is used when run on ABCL. The behavior of `compareAndSet` (https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReference.html#compareAndSet-V-V-) is similar to that of the other CL implementations. But I can also have a look at this once we have this done.

Reply:
> There is a special file `atomic-abcl.lisp` that is used when run on ABCL. The behavior of `compareAndSet` (https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicReference.html#compareAndSet-V-V-) is similar to that of the other CL implementations. But I can also have a look at this once we have this done.

I'm sorry I have no time to install ABCL today. I went to the hospital today, my son's ill.

Reply:
Don't worry. No rush. I will can take care of the ABCL stuff.